packages/salesforcedx-vscode-apex-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -426,6 +422,166 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
+      'sfdx_update_checkpoints_in_org'
+    );
+    // Wrap everything in a try/finally to ensure creatingCheckpoints gets set to false
+    try {
+      // The lock is necessary here to prevent the user from deleting the underlying breakpoint
+      // attached to the checkpoint while they're being uploaded into the org.
+      await lock.acquire(CHECKPOINTS_LOCK_STRING, async () => {
+        writeToDebuggerOutputWindow(
+          `${nls.localize('long_command_start')} ${localizedProgressMessage}`
+        );
+        await vscode.window.withProgress(
+          {
+            location: vscode.ProgressLocation.Notification,
+            title: localizedProgressMessage,
+            cancellable: false
+          },
+          async (progress, token) => {
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_org_info'
+              )}`
+            );
+            progress.report({
+              increment: 0,
+              message: localizedProgressMessage
+            });
+            const orgInfoRetrieved: boolean = await checkpointService.retrieveOrgInfo();
+            if (!orgInfoRetrieved) {
+              updateError = true;
+              return false;
+            }
+
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_source_line_info'
+              )}`
+            );
+            progress.report({
+              increment: 20,
+              message: localizedProgressMessage
+            });
+            const sourceLineInfoRetrieved: boolean = await retrieveLineBreakpointInfo();
+            // If we didn't get the source line information that'll be reported at that time, just return
+            if (!sourceLineInfoRetrieved) {
+              updateError = true;
+              return false;
+            }
+
+            // There can be a max of five active checkpoints
+            if (!checkpointService.hasFiveOrLessActiveCheckpoints(true)) {
+              updateError = true;
+              return false;
+            }
+
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_setting_typeref'
+              )}`
+            );
+            progress.report({
+              increment: 50,
+              message: localizedProgressMessage
+            });
+            // For the active checkpoints set the typeRefs using the source/line info
+            if (!setTypeRefsForEnabledCheckpoints()) {
+              updateError = true;
+              return false;
+            }
+
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_clearing_existing_checkpoints'
+              )}`
+            );
+            progress.report({
```

Comment:
I'm not familiar with progress.report().  Is `increment` "increment by" or "increment to"?
I looks like it's probably "increment to".  If that's the case, the progress is being set to 50% on lines 506 and 521.

Reply:
Good q, I had no idea but looking at the docs it appears to sum the increment
```
To report discrete progress, use `increment` to indicate how much work has been completed. Each call with
         * a `increment` value will be summed up and reflected as overall progress until 100% is reached (a value of
         * e.g. `10` accounts for `10%` of work done).
         * 
         * 
```
From `packages/salesforcedx-utils-vscode/node_modules/@types/vscode/index.d.ts`

Honestly I don't ever recall progress ever being reported for any of our operations so it makes me wonder if it's even working as expected in the product today.